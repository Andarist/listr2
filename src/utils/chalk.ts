import chalkDefault from 'chalk'

let chalk: chalkDefault.Chalk

/* istanbul ignore if */
if (process.env?.LISTR_DISABLE_COLOR !== '1' && chalkDefault.supportsColor) {
  chalk = new chalkDefault.Instance()
} else {
  chalk = new chalkDefault.Instance({ level: 0 })
}

/** A custom chalk instance to disable coloring compeletely. */
export default chalk
