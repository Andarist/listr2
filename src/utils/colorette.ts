import { createColors } from 'colorette'

export default createColors({ useColor: process.env?.LISTR_DISABLE_COLOR !== '1' })
