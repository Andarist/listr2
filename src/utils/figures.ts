import { isUnicodeSupported } from './is-unicode-supported'

const FIGURES_MAIN = {
  warning: '⚠',
  cross: '✖',
  arrowDown: '↓',
  tick: '✔',
  arrowRight: '→',
  pointer: '❯',
  checkboxOn: '☒',
  arrowLeft: '←',
  squareSmallFilled: '◼',
  pointerSmall: '›'
}

const FIGURES_FALLBACK = {
  ...FIGURES_MAIN,
  warning: '‼',
  cross: '×',
  tick: '√',
  pointer: '>',
  checkboxOn: '[×]',
  squareSmallFilled: '■'
}

export const figures = isUnicodeSupported() ? FIGURES_MAIN : FIGURES_FALLBACK
