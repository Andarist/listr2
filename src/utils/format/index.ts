export * from './colorette'
export * from './indent-string'
export * from './figures'
export * from './parse-time'
export * from './splat'
export * from './timestamp'
