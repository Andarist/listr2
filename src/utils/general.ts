import * as clone from 'clone'

/**
 * Deep clones a object in the most easiest manner.
 */
export function cloneObject<T extends Record<PropertyKey, any>> (obj: T): T {
  return clone(obj)
}
