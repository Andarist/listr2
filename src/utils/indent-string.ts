export function indentString (string: string, count: number): string {
  return string.replace(/^(?!\s*$)/gm, ' '.repeat(count))
}