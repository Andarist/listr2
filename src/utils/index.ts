export * from './environment'
export * from './format'
export * from './logger'
export * from './process-output'
export * from './ui'
export * from './assert'
export * from './clone'
export * from './concurrency'
export * from './delay'
export * from './uuid'
