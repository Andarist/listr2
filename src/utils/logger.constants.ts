/** Default loglevels for the logger */
export enum LogLevels {
  silent = 'silent',
  fail = 'fail',
  skip = 'skip',
  success = 'success',
  data = 'data',
  start = 'start',
  title = 'title',
  retry = 'retry',
  rollback = 'rollback'
}
