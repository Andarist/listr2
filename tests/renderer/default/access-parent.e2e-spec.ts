import { expectProcessOutputToMatchSnapshot, mockProcessOutput, unmockProcessOutput } from '@tests/utils'
import type { MockProcessOutput } from '@tests/utils'

import { Listr } from '@root'

describe('default renderer: parent', () => {
  const output: MockProcessOutput = {} as MockProcessOutput

  process.stdout.isTTY = true

  beforeEach(async () => {
    mockProcessOutput(output)
  })

  afterEach(async () => {
    unmockProcessOutput(output)
    jest.clearAllMocks()
  })

  // nvFwliir81dQSzoa7xPXaq0E1lauH674
  it('should be able to change the title of parent from subtask', async () => {
    await new Listr(
      [
        {
          title: 'This task will execute.',
          task: (_, task): Listr =>
            task.newListr((parent) => [
              {
                title: 'This is a subtask.',
                task: async (): Promise<void> => {
                  parent.title = 'Title changed from subtask.'
                }
              },

              {
                title: 'This is a subtask.',
                task: async (): Promise<void> => Promise.resolve()
              }
            ])
        }
      ],
      {
        concurrent: false,
        rendererOptions: { lazy: true }
      }
    ).run()

    expectProcessOutputToMatchSnapshot(output, 'nvFwliir81dQSzoa7xPXaq0E1lauH674')
  })
})
