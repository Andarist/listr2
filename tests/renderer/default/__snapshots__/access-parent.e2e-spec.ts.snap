// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default renderer: parent should be able to change the title of parent from subtask: nvFwliir81dQSzoa7xPXaq0E1lauH674-exit 1`] = `[]`;

exports[`default renderer: parent should be able to change the title of parent from subtask: nvFwliir81dQSzoa7xPXaq0E1lauH674-stderr 1`] = `[]`;

exports[`default renderer: parent should be able to change the title of parent from subtask: nvFwliir81dQSzoa7xPXaq0E1lauH674-stdout 1`] = `
[
  "◼ This task will execute.
",
  "[2K[1A[2K[G❯ This task will execute.
",
  "[2K[1A[2K[G❯ This task will execute.
  ◼ This is a subtask.
",
  "[2K[1A[2K[1A[2K[G❯ This task will execute.
  ◼ This is a subtask.
  ◼ This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ This task will execute.
  ❯ This is a subtask.
  ◼ This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ Title changed from subtask.
  ❯ This is a subtask.
  ◼ This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ Title changed from subtask.
  ✔ This is a subtask.
  ◼ This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ Title changed from subtask.
  ✔ This is a subtask.
  ❯ This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G❯ Title changed from subtask.
  ✔ This is a subtask.
  ✔ This is a subtask.
",
  "[2K[1A[2K[1A[2K[1A[2K[G✔ Title changed from subtask.
",
  "[2K[1A[2K[G",
  "✔ Title changed from subtask.
",
]
`;
